### ISSUE TITLE
Add all option to iegitaddmd.ado

### ISSUE THREAD

**Issue by bbdaniels** [worldbank/ietoolkit#101]
Add option to add readme placeholder to _every_ subfolder. Syntax would be:
```
(line 7) syntax , folder(string) [all]
(line 25) if (`"`flist'`dlist'`olist'"' == "") | ("`all'" == "all") { 
```

**Comment by luizaandrade** [comment]
I like that, goes with our suggestion to not delete it even if the folder is populated, and gives people the choice to do that. What do you think, @kbjarkefur?

**Comment by bbdaniels** [comment]
I added this option as suggested to 0029ef25164ea5452f22979c13b1f95a9bfaea25 and corrected what I believe was an error in the recursive call in line – please review in line 37.

**Comment by luizaandrade** [comment]
I think both  commits 7677e6af7079236b95246327f52e7ca6b583677f and 0029ef25164ea5452f22979c13b1f95a9bfaea25 depend on settling which is the best way to use read me files in folder. I'll merge it once we agreed on that.

**Comment by kbjarkefur** [comment]
I think that is a great addition!

Another reason for it as I edited the gitignore template is that if you have a data folder with data files that are ignored, then **iegitaddmd** does not add a *README.md* file but the folder will not be synced as the files in that folder will not be synced.

An extension of this is (that I do not think we should spend time on until we know that this command will be used frequently) would be for **iegitaddmd** to be able to tell if a folder will be synced or not. As in see what files there is in a folder and add *README.md* if all files in a folder will be ignored but not if there are some files that will be ignored.

**Comment by kbjarkefur** [comment]
I added a test in 8521e51 that tests if a *README.md* file already exist before creating it. This could happen when using the [all] option multiple times (could happen if **iegitaddmd** is used after adding new folders) or if [all] is used on a folder that already has a *README.md* created manually. 

The test results in a *README.md* file not being created if it already exists.

**Comment by kbjarkefur** [comment]
I did a small stupid error that I fixed in 2771bcb

**Comment by kbjarkefur** [comment]
I should perhaps test the command better before pushing. We forgot to pass the value of **all** to recursive calls of the command. Before afaf99b it was only applied to the top folder and not subfolders.

I don't think there is anything else 😟
